src/python/grpcio_tests/tests/health_check/_health_servicer_test.py

```diff
@@ -27,49 +27,70 @@
+"""Tests of grpc.health.v1.health."""
 
 import unittest
 
-from grpc_health.health.v1 import health
-from grpc_health.health.v1 import health_pb2
+import grpc
+from grpc.framework.foundation import logging_pool
+from grpc.health.v1 import health
+from grpc.health.v1 import health_pb2
+
+from tests.unit.framework.common import test_constants
 
 
 class HealthServicerTest(unittest.TestCase):
 
   def setUp(self):
-    self.servicer = health.HealthServicer()
-    self.servicer.set('', health_pb2.HealthCheckResponse.SERVING)
-    self.servicer.set('grpc.test.TestServiceServing',
+    servicer = health.HealthServicer()
+    servicer.set('', health_pb2.HealthCheckResponse.SERVING)
+    servicer.set('grpc.test.TestServiceServing',
                       health_pb2.HealthCheckResponse.SERVING)
-    self.servicer.set('grpc.test.TestServiceUnknown',
+    servicer.set('grpc.test.TestServiceUnknown',
                       health_pb2.HealthCheckResponse.UNKNOWN)
-    self.servicer.set('grpc.test.TestServiceNotServing',
+    servicer.set('grpc.test.TestServiceNotServing',
                       health_pb2.HealthCheckResponse.NOT_SERVING)
+    server_pool = logging_pool.pool(test_constants.THREAD_CONCURRENCY)
+    self._server = grpc.server(server_pool)
+    port = self._server.add_insecure_port('[::]:0')
+    health_pb2.add_HealthServicer_to_server(servicer, self._server)
+    self._server.start()
+
+    channel = grpc.insecure_channel('localhost:%d' % port)
+    self._stub = health_pb2.HealthStub(channel)
 
   def test_empty_service(self):
     request = health_pb2.HealthCheckRequest()
-    resp = self.servicer.Check(request, None)
+    resp = self._stub.Check(request)
     self.assertEqual(resp.status, health_pb2.HealthCheckResponse.SERVING)
 
   def test_serving_service(self):
     request = health_pb2.HealthCheckRequest(
         service='grpc.test.TestServiceServing')
-    resp = self.servicer.Check(request, None)
+    resp = self._stub.Check(request)
     self.assertEqual(resp.status, health_pb2.HealthCheckResponse.SERVING)
 
   def test_unknown_serivce(self):
     request = health_pb2.HealthCheckRequest(
         service='grpc.test.TestServiceUnknown')
-    resp = self.servicer.Check(request, None)
+    resp = self._stub.Check(request)
     self.assertEqual(resp.status, health_pb2.HealthCheckResponse.UNKNOWN)
 
   def test_not_serving_service(self):
     request = health_pb2.HealthCheckRequest(
         service='grpc.test.TestServiceNotServing')
-    resp = self.servicer.Check(request, None)
+    resp = self._stub.Check(request)
     self.assertEqual(resp.status, health_pb2.HealthCheckResponse.NOT_SERVING)
 
+  def test_not_found_service(self):
+    request = request = health_pb2.HealthCheckRequest(
+        service='not-found')
+    with self.assertRaises(grpc.RpcError) as context:
+      resp = self._stub.Check(request)
+  
+    print(context.exception.details())
+    self.assertEqual(context.exception.code(), grpc.StatusCode.NOT_FOUND)
```

Comment:
Python doesn't define an order like the unit test libraries of other languages do but throughout our codebase we've tried to maintain `<expected value>, <value obtained from system under test>` for the arguments to `assertEqual`. Please continue that?

On comment proofreading: yes, this means changing otherwise-untouched lines in this test.